src/pymatgen/analysis/magnetism/heisenberg.py

```diff
@@ -320,6 +320,8 @@ def _get_exchange_df(self):
 
                 # Ignore the row if it is a duplicate to avoid singular matrix
                 # Create a temporary DataFrame with the new row
+                ex_mat = ex_mat.dropna(how="all", axis=1)
```

Comment:
I'm not 100% sure about this change (regarding the axis), but it should be the columns from the warning message:
```
  /Users/yang/developer/pymatgen/src/pymatgen/analysis/magnetism/heisenberg.py:325: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
    temp_df = pd.concat([ex_mat, ex_row], ignore_index=True)

```